osf/models/mixins.py

```diff
@@ -786,6 +787,8 @@ class ContributorMixin(models.Model):
     class Meta:
         abstract = True
 
+    DEFAULT_CONTRIBUTOR_PERMISSIONS = 'write'
```

Comment:
Most of the ContributorMixin changes involve moving Preprint model contributor methods to this mixin, now that both Preprints and Nodes are using guardian.

Also removes some special casing to handle nodes versus preprints.